builtin/providers/azure/resource_azure_instance.go

```diff
@@ -594,6 +608,37 @@ func resourceAzureInstanceDelete(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	log.Printf("[INFO] Waiting for lease on instance '%s''s disk blob to be lifted.", name)
+
+	// in order to avoid `terraform taint`-like problems in which the instance
+	// is deleted and re-created so fast the previous storage blob which held
+	// the image doesn't manage to get deleted (despite it being in a
+	// 'deleting' state) and a lease conflict occurs over it, we must ensure
+	// the blob got completely deleted as well:
+	storName := d.Get("storage_service_name").(string)
+	blobClient, err := azureClient.getStorageServiceBlobClient(storName)
+	if err != nil {
+		return err
+	}
+
+	// unfortunately, there's no better way of doing this except
+	// polling continuously until it goes away:
```

Comment:
Can we toss an outer limit timeout on this?

We have a `resource.Retry` helper that's commonly used elsewhere for this purpose:

https://github.com/hashicorp/terraform/blob/master/helper/resource/wait.go#L12-L36

Example usage:

https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_autoscaling_group.go#L374-L394